java/lib/ghsl/CommandInjectionRuntimeExec.qll

```diff
@@ -4,124 +4,111 @@ import semmle.code.java.dataflow.DataFlow
-            and sink.asExpr().getType() instanceof Array
-        )
+    predicate isSink(DataFlow::Node sink) {
+      exists(RuntimeExecMethod method, MethodCall call |
+        call.getMethod() = method and
+        sink.asExpr() = call.getArgument(0) and
+        sink.asExpr().getType() instanceof Array
+      )
     }
 
-    override
-    predicate
-    isSanitizer(DataFlow::Node node) {
-        node.asExpr().getFile().isSourceFile() and
-        (
-            node instanceof AssignToNonZeroIndex
-            or node instanceof ArrayInitAtNonZeroIndex
-            or node instanceof StreamConcatAtNonZeroIndex
-            or node.getType() instanceof PrimitiveType
-            or node.getType() instanceof BoxedType
-        )
+    predicate isBarrier(DataFlow::Node node) {
+      node.asExpr().getFile().isSourceFile() and
+      (
+        node instanceof AssignToNonZeroIndex or
+        node instanceof ArrayInitAtNonZeroIndex or
+        node instanceof StreamConcatAtNonZeroIndex or
+        node.getType() instanceof PrimitiveType or
+        node.getType() instanceof BoxedType
+      )
     }
-}
+  }
 
-abstract class Source extends DataFlow::Node {
-    Source() {
-        this = this
-    }
-}
+  module RuntimeExecFlow = TaintTracking::Global<RuntimeExecConfiguration>;
 
+  import RuntimeExecFlow::PathGraph
+}
 
 // taint flow from user data to args of the command
-class ExecTaintConfiguration2 extends TaintTracking::Configuration {
-    ExecTaintConfiguration2() { this = "ExecTaintConfiguration2" }
-
-    override
-    predicate
-    isSource(DataFlow::Node source) {
-        source instanceof Source
+module ExecTaint {
+  module ExecTaintConfiguration implements DataFlow::ConfigSig {
+    predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    predicate isSink(DataFlow::Node sink) {
+      exists(RuntimeExecMethod method, MethodCall call, int index |
+        call.getMethod() = method and
+        sink.asExpr() = call.getArgument(index) and
+        sink.asExpr().getType() instanceof Array
+      )
     }
 
-    override
-    predicate
-    isSink(DataFlow::Node sink) {
-        exists(RuntimeExecMethod method, MethodAccess call, int index |
-            call.getMethod() = method
-            and sink.asExpr() = call.getArgument(index)
-            and sink.asExpr().getType() instanceof Array
-        )
+    predicate isBarrier(DataFlow::Node node) {
+      node.asExpr().getFile().isSourceFile() and
+      (
+        node.getType() instanceof PrimitiveType or
+        node.getType() instanceof BoxedType
+      )
     }
+  }
 
-    override
-    predicate
-    isSanitizer(DataFlow::Node node) {
-        node.asExpr().getFile().isSourceFile() and
-        (
-            node.getType() instanceof PrimitiveType
-            or node.getType() instanceof BoxedType
-        )
-    }
-}
+  module ExecTaintFlow = TaintTracking::Global<ExecTaintConfiguration>;
 
+  import ExecTaintFlow::PathGraph
```

Comment:
I think this is not necessary. Only to bring in the edges predicate necessary for path-problems